### PR TITLE
Changing Eigen Library

### PR DESCRIPTION
**Pull request by jpatch22**
### Review Checklist

**_It is the reviewers responsibility to also make sure every item here has been covered_**

- [ ] **Function & Class comments**: All function definitions (usually in the `.h` file) should have a javadoc style comment at the start of them. For examples, see the functions defined in `thunderbots/software/geom`. Similarly, all classes should have an associated Javadoc comment explaining the purpose of the class.
- [ ] **Remove all commented out code**
- [ ] **Remove extra print statements**: for example, those just used for testing
- [ ] **Resolve all TODO's**: All `TODO` (or similar) statements should either be completed or associated with a github issue
- [ ] **Justify drops in code coverage**: If this PR results in a non-trivial drop in code coverage (a bot should post a coverage diagram as a comment), please justify why we can't test the code that's not covered.

### CODE REVIEW

File: src/WORKSPACE [hunk 1]
```diff
@@ -73,8 +73,8 @@ boost_deps()
 new_git_repository(
     name = "eigen",
     build_file = "@//external:eigen.BUILD",
-    commit = "21ae2afd4edaa1b69782c67a54182d34efe43f9c",
-    remote = "https://gitlab.com/libeigen/eigen.git",
+    commit = "2bf07fa5b5034ed5f5b1839fea40a2581946f6fa",
```
Comment (jonathanlew) [inline src/WORKSPACE:76]: Changing this commit might be causing the simulated tests to fail, @akhilveeraghanta or someone with ubuntu 20, can you try locally? (easiest might be to revert)

**Review by jonathanlew — CHANGES_REQUESTED** [review]
Thanks for fixing this dependency. Make sure to run the formatting script, i.e. [formatting_scripts/fix_formatting.sh](https://github.com/UBC-Thunderbots/Software/blob/master/formatting_scripts/fix_formatting.sh)

**Review by akhilveeraghanta — COMMENTED** [review]
It looks like an extra change was accidentally added, to `src/software/ai/hl/stp/play/penalty_kick_enemy_play_test.cpp`

make sure to revert that and run `./environemnt_setup/fix_formatting.sh`

and then it should be good to go!

File: src/WORKSPACE (hunk 1 above)
Comment (akhilveeraghanta) [reply src/WORKSPACE:76]: Yeah its busted for me too, since its a mirror I wonder if the old commit will just work, worth a shot

**Comment by jonathanlew** [comment]
Since this was a critical fix, it was merged in as part of #2271

**Pull request closed without merging**